client/config/config.go

```diff
@@ -347,16 +347,23 @@ func (c *Config) ReadStringListToMapDefault(key, defaultValue string) map[string
 	return list
 }
 
-// TLSConfig returns a TLSUtil Config based on the client configuration
+// TLSConfiguration returns a TLSUtil Config based on the existing client
+// configuration
 func (c *Config) TLSConfiguration() *tlsutil.Config {
-	tlsConf := &tlsutil.Config{
+	return c.NewTLSConfiguration(c.TLSConfig)
+}
+
+// NewTLSConfiguration returns a TLSUtil Config for a new TLS config object
+// This allows a TLSConfig object to be created without first explicitely
+// setting it
+func (c *Config) NewTLSConfiguration(tlsConfig *config.TLSConfig) *tlsutil.Config {
```

Comment:
This shouldn't be a method on the Config as it is using the parameter. Not sure I understand the need for these two to be separate